activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -0,0 +1,389 @@
+          when String
+            redis.eval DELETE_GLOB_LUA, [], [namespaced_key(matcher, options)]
+          else
+            raise ArgumentError, "Only Redis glob strings are supported: #{matcher.inspect}"
+          end
+        end
+      end
+
+      # Cache Store API implementation.
+      #
+      # Increment a cached value. This method uses the Redis incr atomic
+      # operator and can only be used on values written with the :raw option.
+      # Calling it on a value not stored with :raw will initialize that value
+      # to zero.
+      #
+      # Failsafe: Raises errors.
+      def increment(name, amount = 1, options = nil)
+        instrument :increment, name, amount: amount do
+          redis.incrby normalize_key(name, options), amount
+        end
+      end
+
+      # Cache Store API implementation.
+      #
+      # Decrement a cached value. This method uses the Redis decr atomic
+      # operator and can only be used on values written with the :raw option.
+      # Calling it on a value not stored with :raw will initialize that value
+      # to zero.
+      #
+      # Failsafe: Raises errors.
+      def decrement(name, amount = 1, options = nil)
+        instrument :decrement, name, amount: amount do
+          redis.decrby normalize_key(name, options), amount
+        end
+      end
+
+      # Cache Store API implementation.
+      #
+      # Removes expired entries. Not supported.
+      def cleanup(options = nil)
+        super
+      end
+
+      # Clear the entire cache on all Redis servers. Safe to use on
+      # shared servers if the cache is namespaced.
+      #
+      # Failsafe: Raises errors.
+      def clear(options = nil)
+        if namespace = merged_options(options)[namespace]
+          delete_matched "*", namespace: namespace
+        else
+          failsafe :clear do
+            redis.flushdb
+          end
+        end
+      end
+
+      private
+        # Store provider interface:
+        # Read an entry from the cache.
+        def read_entry(key, options = nil)
+          failsafe :read_entry do
+            deserialize_entry redis.get(key)
+          end
+        end
+
+        def read_multi_mget(*names)
+          options = names.extract_options!
+          options = merged_options(options)
+
+          keys_to_names = names.map { |name| [ normalize_key(name, options), name ] }.to_h
+          values = redis.mget(*keys_to_names.keys)
+
+          keys_to_names.zip(values).each_with_object({}) do |((key, name), value), results|
+            if value
+              entry = deserialize_entry(value)
+              unless entry.nil? || entry.expired? || entry.mismatched?(normalize_version(key, options))
+                results[name] = entry.value
+              end
+            end
+          end
+        end
+
+        # For compat with the wexpected options hash argument.
+        def write_entry(key, entry, options = nil)
+          write_entry_kwargs key, entry, **options
+        end
```

Comment:
Shim method specifically to move from old `options` style API to new kwarg API.

Allows cache store superclass to continue using old options API for existing cache stores while we jump ahead to keyword args style.

Reply:
Don't these have identical arities? :confused:

Reply:
Yes indeed. I could've sworn I hit some issue with supering from `LocalCacheWithRaw#write_entry` with diff method signatures, but I can't tickle it now.